sentry-android-replay/src/main/java/io/sentry/android/replay/viewhierarchy/ComposeViewHierarchyNode.kt

```diff
@@ -83,16 +109,53 @@ internal object ComposeViewHierarchyNode {
             _rootCoordinates = WeakReference(node.coordinates.findRootCoordinates())
         }
 
-        val semantics = node.collapsedSemantics
         val visibleRect = node.coordinates.boundsInWindow(_rootCoordinates?.get())
+        val semantics: SemanticsConfiguration?
+
+        try {
+            semantics = retrieveSemanticsConfiguration(node)
+        } catch (t: Throwable) {
+            if (!semanticsRetrievalErrorLogged) {
+                semanticsRetrievalErrorLogged = true
+                options.logger.log(
+                    SentryLevel.ERROR,
+                    t,
+                    """
+                Error retrieving semantics information from Compose tree. Most likely you're using
+                an unsupported version of androidx.compose.ui:ui. The supported
+                version range is 1.5.0 - 1.8.0.
+                If you're using a newer version, please open a github issue with the version
+                you're using, so we can add support for it.
+                    """.trimIndent()
+                )
+            }
+
+            // If we're unable to retrieve the semantics configuration
```

Comment:
I'm thinking if we should check if any masking is actually enabled? Thinking of a case where nothing should be masked, but we still would weirdly mask something when failed to retrieve the semantics.

Reply:
(maskAllText and maskAllImages, that is)

Reply:
I'm wondering if we should even process the view hierarchy if no masking is active? I'll get this merged now, as by default masking is active anyway. We can always improve on top.